### PR TITLE
Make enclave provider interfaces internal

### PR DESCRIPTION
**Pull request by David-Engel**

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.NetCoreApp.cs [hunk 1]
```diff
@@ -86,13 +86,13 @@ internal abstract class VirtualizationBasedSecurityEnclaveProviderBase : Enclave
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.NetCoreApp.cs:86]: #region Internal methods

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.cs [hunk 2]
```diff
@@ -65,13 +65,13 @@ internal class AzureAttestationEnclaveProvider : EnclaveProviderBase
         #region Public methods
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.cs:65]:
```suggestion
        #region Internal methods
```

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.cs [hunk 3]
```diff
@@ -86,14 +86,14 @@ internal abstract class VirtualizationBasedSecurityEnclaveProviderBase : Enclave
```
Comment (karinazhou) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.cs:86]: Line 85: #region Internal methods

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.NetCoreApp.cs (hunk 1 above)
Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/VirtualSecureModeEnclaveProviderBase.NetCoreApp.cs:86]: Fixed

**Pull request merged**